### ISSUE TITLE
DBInstance cannot be created

### ISSUE THREAD

**Issue by NikitaIvankov1992** [aws-controllers-k8s/community#1727]
**Describe the bug**
We're trying to create AWS Aurora DBCluster and few DBInstance along with DBClusterParameterGroup and DBParameterGroup
Everything besides DBInstance are created fine, but DBInstance gets stuck with the following error
```
status:
  conditions:
  - lastTransitionTime: "2023-03-14T07:39:45Z"
    message: the referenced resource is not synced yet. resource:DBParameterGroup,
      namespace:default, name:test-rds-2-aurora-postgresql14
    status: Unknown
    type: ACK.ReferencesResolved
``` 
If checking status of the DBParameterGroup
```
  ackResourceMetadata:
    arn: arn:aws:rds:eu-central-1:810640839231:pg:test-rds-2-aurora-postgresql14
    ownerAccountID: "810640839231"
    region: eu-central-1
  conditions:
  - message: parameter group just got created
    status: "True"
    type: ACK.Recoverable
  - lastTransitionTime: "2023-03-14T07:03:46Z"
    message: Unable to determine if desired resource state matches latest observed
      state
    reason: parameter group just got created
    status: Unknown
    type: ACK.ResourceSynced
```
And in the logs of controller I can see many errors like the following one
```
2023-03-14T07:39:45.806Z        ERROR   controller.dbinstance   Reconciler error        {"reconciler group": "rds.services.k8s.aws", "reconciler kind": "DBInstance", "name": "test-rds-2-0", "namespace": "default", "error": "the referenced resource is not synced yet. resource:DBParameterGroup, namespace:default, name:test-rds-2-aurora-postgresql14"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
```
but seems like no errors related to creation of DBParameterGroup
```
2023-03-14T07:17:53.936Z        INFO    ackrt   desired resource state has changed      {"account": "810640839231", "role": "", "region": "eu-central-1", "kind": "DBParameterGroup", "namespace": "default", "name": "test-rds-2-aurora-postgresql14", "is_adopted": false, "generation": 1, "diff": [{"Path":{"Parts":["Spec","ParameterOverrides"]},"A":null,"B":{}}]}
2023-03-14T07:17:53.936Z        INFO    ackrt   updated resource        {"account": "810640839231", "role": "", "region": "eu-central-1", "kind": "DBParameterGroup", "namespace": "default", "name": "test-rds-2-aurora-postgresql14", "is_adopted": false, "generation": 1}
```
Also seems like there is no error in AWS API
```
{
    "eventTime": "2023-03-14T07:03:46Z",
    "eventSource": "rds.amazonaws.com",
    "eventName": "CreateDBParameterGroup",
    "awsRegion": "eu-central-1",
    "sourceIPAddress": "35.159.44.187",
    "userAgent": "aws-controllers-k8s/-v1.0.0 (GitCommit/2afec4930746b7a473f8e346fb49f61a29b7c3b5; BuildDate/2023-02-14T10:55; CRDKind/; CRDVersion/) aws-sdk-go/1.44.93 (go1.19.5; linux; amd64)",
    "requestParameters": {
        "dBParameterGroupName": "test-rds-2-aurora-postgresql14",
        "dBParameterGroupFamily": "aurora-postgresql14",
        "description": "Parameter group for test-rds-2 cluster instances",
        "tags": [
            {
                "key": "services.k8s.aws/controller-version",
                "value": "rds-v1.0.0"
            },
            {
                "key": "services.k8s.aws/namespace",
                "value": "default"
            }
        ]
    },
    "responseElements": {
        "dBParameterGroupName": "test-rds-2-aurora-postgresql14",
        "dBParameterGroupFamily": "aurora-postgresql14",
        "description": "Parameter group for test-rds-2 cluster instances",
        "dBParameterGroupArn": "arn:aws:rds:eu-central-1:810640839231:pg:test-rds-2-aurora-postgresql14"
    },
    "requestID": "708763b1-573a-4d90-9e4d-ce0ac1d2eb41",
    "eventID": "5c8db8d2-4dab-4eb7-99e9-2f530682ce7c",
    "readOnly": false,
    "eventType": "AwsApiCall",
    "managementEvent": true,
    "recipientAccountId": "810640839231",
    "eventCategory": "Management",
    "tlsDetails": {
        "tlsVersion": "TLSv1.2",
        "cipherSuite": "ECDHE-RSA-AES128-GCM-SHA256",
        "clientProvidedHostHeader": "rds.eu-central-1.amazonaws.com"
    }
}
```

**Steps to reproduce**
We use following Helm resources
Cluster
```
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBCluster
metadata:
  annotations:
    meta.helm.sh/release-name: test-rds-2
    meta.helm.sh/release-namespace: default
  creationTimestamp: "2023-03-14T07:03:45Z"
  finalizers:
  - finalizers.rds.services.k8s.aws/DBCluster
  generation: 2
  labels:
    app.kubernetes.io/managed-by: Helm
  name: test-rds-2
  namespace: default
  resourceVersion: "778228942"
  uid: 39c87efe-be27-411f-8688-59dd3801c20c
spec:
  allocatedStorage: 1
  autoMinorVersionUpgrade: false
  availabilityZones:
  - eu-central-1b
  - eu-central-1c
  - eu-central-1a
  backupRetentionPeriod: 21
  copyTagsToSnapshot: false
  databaseName: dev
  dbClusterIdentifier: test-rds-2
  dbClusterParameterGroupRef:
    from:
      name: test-rds-2-aurora-postgresql14
  dbSubnetGroupName: vpc-platform-dev
  deletionProtection: true
  engine: aurora-postgresql
  engineMode: provisioned
  engineVersion: "14.6"
  kmsKeyID: arn:aws:kms:eu-central-1:810640839231:key/099e7530-c5d6-428a-93e8-a10c8c0de53f
  masterUserPassword:
    key: password
    name: test-rds-2
    namespace: default
  masterUsername: root
  networkType: IPV4
  port: 5432
  preferredBackupWindow: 01:09-01:39
  preferredMaintenanceWindow: fri:23:37-sat:00:07
  storageEncrypted: true
  vpcSecurityGroupRefs:
  - from:
      name: test-rds-2
```

Instances
```
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBInstance
metadata:
  annotations:
    meta.helm.sh/release-name: test-rds-2
    meta.helm.sh/release-namespace: default
  creationTimestamp: "2023-03-14T07:03:45Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
  name: test-rds-2-0
  namespace: default
  resourceVersion: "778239462"
  uid: 777e1849-0e2c-42a6-b059-e1385a62cbc3
spec:
  dbClusterIdentifier: test-rds-2
  dbInstanceClass: db.t3.medium
  dbInstanceIdentifier: test-rds-2-0
  dbParameterGroupRef:
    from:
      name: test-rds-2-aurora-postgresql14
  deletionProtection: true
  engine: aurora-postgresql
```

DBParameterGroup
```
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBParameterGroup
metadata:
  annotations:
    meta.helm.sh/release-name: test-rds-2
    meta.helm.sh/release-namespace: default
  creationTimestamp: "2023-03-14T07:03:46Z"
  finalizers:
  - finalizers.rds.services.k8s.aws/DBParameterGroup
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
  name: test-rds-2-aurora-postgresql14
  namespace: default
  resourceVersion: "778221334"
  uid: a00aaeea-54f2-4f1d-ab5e-04faf55ff2f2
spec:
  description: Parameter group for test-rds-2 cluster instances
  family: aurora-postgresql14
  name: test-rds-2-aurora-postgresql14
```

DBClusterParameterGroup
```
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBClusterParameterGroup
metadata:
  annotations:
    meta.helm.sh/release-name: test-rds-2
    meta.helm.sh/release-namespace: default
  creationTimestamp: "2023-03-14T07:03:45Z"
  finalizers:
  - finalizers.rds.services.k8s.aws/DBClusterParameterGroup
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
  name: test-rds-2-aurora-postgresql14
  namespace: default
  resourceVersion: "778228930"
  uid: 542b1a44-1cb6-428c-ba4e-5463eb7698c9
spec:
  description: Cluster parameter group for the test-rds-2 cluster
  family: aurora-postgresql14
  name: test-rds-2-aurora-postgresql14
```

**Expected outcome**
We expect to create DBCluster and DBInstances

* Kubernetes version 1.24
* Using EKS (yes/no), if so version?  eks.4
* AWS service targeted (S3, RDS, etc.) RDS

**Comment by jaypipes** [comment]
@NikitaIvankov1992 OK, thanks for that. Please check in a few hours, as the default requeue is 10 hours. In the meantime, I will check into why the DBParameterGroup resource isn't behaving the same way as the DBClusterParameterGroup resource.

**Comment by A-Hilaly** [comment]
@jaypipes @NikitaIvankov1992 we could also leverage [`defaultResyncPeriod`](https://github.com/aws-controllers-k8s/rds-controller/blob/main/helm/values.yaml#L89) to reduce the wait time here?

**Comment by jaypipes** [comment]
> I think I can try to update controller with [defaultResyncPeriod](https://github.com/aws-controllers-k8s/rds-controller/blob/main/helm/values.yaml#L89) but in general it shouldn't be like this, right?

Correct. The DBParameterGroup should behave like DBClusterParameterGroup and get into a Synced status quickly.

**Comment by A-Hilaly** [comment]
>I think I can try to update controller with [defaultResyncPeriod](https://github.com/aws-controllers-k8s/rds-controller/blob/main/helm/values.yaml#L89) but in general it shouldn't be like this, right?

@NikitaIvankov1992 `0` means that it will default to 10 hours https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/config/config.go#L162-L167
Ideally we should bring this doc string to the helm chart as well :)

**Comment by A-Hilaly** [comment]
@NikitaIvankov1992 I just sent a [PR to release `v1.1.0`](https://github.com/aws-controllers-k8s/rds-controller/pull/139) - looks like we never released a version with `--resync-*` flags.
In the meantime, a hack to trigger a resync would be to scale down and up the deployment replicas (1 to 0 then 0 to 1) or to kill the running pod and let the deployment controller restart it. In a nutshell, a controller can do a fresh reconciliation every time it starts up.

>Unfortunately, I couldn't test if DBParameterGroup got created after 10h of default timeout because it's still init setup for us and we decided to change AWS IAM role name and didn't update k8s SA annotation meanwhile, so the error I got today was

Unfortunately, I can't debug the error here, but clearly, it's an IRSA configuration issue. You might want to set it up from scratch :)

**Comment by jaypipes** [comment]
@NikitaIvankov1992 OK, cool, thanks for following up on this. I'm in the middle of reworking the e2e tests for the RDS controller (see #1726 ). I will be adding additional tests to verify the syned state of parameter groups and try to get to the bottom of this problem. Thanks for your patience!

**Comment by jaypipes** [comment]
@NikitaIvankov1992 working on this right now.

**Comment by jaypipes** [comment]
@NikitaIvankov1992 quick update... I'm still actively working on the reproducible e2e test for this scenario.

**Comment by jaypipes** [comment]
OK, @NikitaIvankov1992 well the good news is I now have a reproducible test case. :) The bad news is that I have not yet figured out what is causing this behaviour. I will continue poking in the controller hook code to see what the issue is and let you know my progress.

**Comment by jaypipes** [comment]
@NikitaIvankov1992 fixed in the above PR.

**Comment by jaypipes** [comment]
@NikitaIvankov1992 the fix is [PR #1612](https://github.com/aws-controllers-k8s/rds-controller/pull/143) and has not yet been merged :)
